### ISSUE TITLE
Unable to assign the process to the job object. Error: The system cannot find the file specified.

### ISSUE THREAD

**Issue by Reko24** [joelverhagen/TorSharp#26]
Error in : 
`await proxy.ConfigureAndStartAsync();
`
> Unable to assign the process to the job object. Error: The system cannot find the file specified.

`

**Comment by joelverhagen** [comment]
I was unable to repro this. Feel free to re-open if you are still having this problem.

**Comment by Sergey1997** [comment]
Have the same issue, seems like issue with workstation.
Just sent code to the server, working fine there.

**Comment by AdamCaviness** [comment]
I have the same issue on one of my machines. Works fine on my laptop but not my desktop. Both are Win10 Enterprise (version 1909).  I figure this has to be a windows feature that isn't installed on my desktop so I tried installing the following individually but none of them correct it:

- Windows Process Activation Service
- Virtual Machine Platform
- Windows Sandbox
- Windows Projected File System
- Guarded Host

I also installed the following on my laptop to match what I have installed on my desktop. None of these caused my laptop to exhibit this issue:
- WSL (linux subsystem)
- Windows containers (windows feature) 

[Setting the TorSharpSettings ToolRunnerType to Simple](https://github.com/joelverhagen/TorSharp/issues/14) does work around the problem but I'd like to understand it and _ideally_ not see privoxy.  

The only remaining stuff on my desktop that I haven't put on my laptop to test are:
- Hyper-V
- Docker for Windows

I didn't install those two because of hassle and cleanup. Also, I suppose I could download the source and debug what the VirtualDesktopToolRunner's processInformation.hProcess is at runtime.

At any rate, this is a really nice project and at least I have a workaround. Thank you!

**Comment by AdamCaviness** [comment]
@joelverhagen  I still receive this error on my desktop. I have updated to Win10 version 2004 and I still get the error. 

The exception is thrown when calling `VirtualDesktopToolRunner.AssociateWithJob()` the second time in `StartAsync()`. There's an argument on `AssociateWithJob` named `throwOnError`. The first call to `AssociateWithJob` is for ctfmonProcess and the `throwOnError` arg is set to false. The second call is for the privoxy and `throwOnError` is set to true. If I set the second call's `throwOnError` to false my app runs privoxy _correctly_. This means that `WindowsApi.SetInformationJobObject()` returns false on my desktop machine but not my laptop but runs it successfully anyway??

I am unable to reopen this issue. I suppose I could submit a pull request but it I'm afraid it would seem rather weak since I'd only be changing the one arg on the second call to `AssociateWithJob()`. Also, I'm not sure that everyone needs this since only a few of us are apparently able to reproduce this so it may be best as setting in `TorSharpSettings`. 

Therefore I'm happy to help in any of the following ways:
1. Let you handle this.
2. Submit a pull request where we always ignore errors on the second `AssociateWithJob()` call for privoxy.
3. Submit a pull request with a new `TorSharpSettings` option to `IgnoreVirtualDesktopErrors` and respect that in `VirtualDesktopToolRunner`.

Thank you!

**Comment by joelverhagen** [comment]
I'm fine with option 3 there. Feel free to send that PR and we'll get it released!

Until I can repro it myself, I cannot do option 1. I am worried about option 2 since with that API call `AssociateWithJob` fails, that means if the TorSharp hosting process (your .NET app) crashes, privoxy or Tor process might live on. The "job" concept allows Windows to couple the privoxy and Tor processes with the parent, handling the clean-up if the parent crashes. At the time I thought this was an important value-add of TorSharp but maybe it's fine to have this option to work around the problem until we understand it more.

Thanks for offering to do the PR!

**Comment by AdamCaviness** [comment]
@joelverhagen  I think it may be best that I not submit a pull request as I described. I found the root cause lies somewhere within the Windows profile. I _think_ it's due to using an Azure Active Directory (work or school) Windows account. Microsoft doesn't consider these "local" windows accounts but they look identical in practice. 

Backstory: I thought I would try one more thing before adding a setting to workaround what only a few people have encountered. One thing that has been bothering me since I installed Windows on this desktop (Enterprise edition) was that when using the Windows installation choice for signing in via AAD (Azure Active Directory) it inserted a space into my Windows username. That had the side effect of bungling the Windows run dialog such that accessing my windows profile path via the run dialog (or explorer in general) _always required wrapping double quotes_. Additionally, just entering %localappdata% in the run dialog behaved the same since Windows expands that to the full path so I'd have to wrap %localappdata% as "%localappdata%" as well. When not using wrapping double quotes, Windows would open the "How do you want to open this file" dialog because the space implies the first string is an application path and the second string is an argument. Another annoyance was after starting Windows two of those "How do you want to open this file" dialogs would open due to auto-starting applications improperly referencing files in my profile without wrapping the path in double quotes. I decided that I'd take a crack at switching to a local windows account (disconnect from AAD) so I could remove the space in the profile name. Windows doesn't make a new profile but reuses it (crazy right?). After naming the local account with no space and _renaming_ the profile folder I was able to run with 'ToolRunnerType' set to `ToolRunnerType.VirtualDesktop`. Why does this fix the problem? My first inclination was that the profile path space was the original problem **but** the Win32 call correctly wraps the path in double quotes so that couldn't be the real problem. To confirm my suspicion of AAD Windows accounts I created a new local Windows user with a space in the username which makes a Windows profile with a space as well. In that new local Windows account I could not reproduce the issue; it ran successfully. 

To recap, a profile path with a space is not a problem but by trying to remove it, I stopped using AAD Windows account and that appears to have corrected it.
